azure-pipelines.yml

```diff
@@ -137,28 +136,51 @@ steps:
   - task: Docker@2
     displayName: Login to temporary pipeline ACR
     inputs:
-      command: login
-      containerRegistry: $(PIPELINE_CONTAINER_REGISTRY)
+      containerRegistry: 'ASO Pipeline ACR'
+      command: 'login'
 
   - task: Docker@2
     displayName: Build and Push Docker Image to temporary ACR for validation
     inputs:
-      command: buildAndPush
-      containerRegistry: $(PIPELINE_CONTAINER_REGISTRY)
-      repository: $(IMAGE_NAME)
-      Dockerfile: "$(System.DefaultWorkingDirectory)/Dockerfile"    
-      buildContext: $(System.DefaultWorkingDirectory)
-      tags: |
-        $(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)
+      containerRegistry: 'ASO Pipeline ACR'
+      repository: '$(IMAGE_NAME)'
+      command: 'buildAndPush'
+      Dockerfile: '$(System.DefaultWorkingDirectory)/Dockerfile'
+      buildContext: '$(System.DefaultWorkingDirectory)'
+      tags: '$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)'
+
+#  - task: Kubernetes@1
+#    displayName: Deploy to AKS - Fetch kubeconfig
+#    inputs:
+#      connectionType: 'Kubernetes Service Connection'
+#      kubernetesServiceEndpoint: 'ASO-AKS'
+#      namespace: $(OPERATOR_NAMESPACE)
+#      command: 'login'
+#      checkLatest: true
 
-  - task: Kubernetes@1
-    displayName: Deploy to AKS - Fetch kubeconfig
+  
+  - task: AzureCLI@2
     inputs:
-      connectionType: 'Kubernetes Service Connection'
-      kubernetesServiceEndpoint: 'ASO-AKS'
-      namespace: $(OPERATOR_NAMESPACE)
-      command: 'login'
-      checkLatest: true
+      azureSubscription: 'ASO Subscription'
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: |
+        az resource list --tag freeforpipeline=true --query '[].{Name:name}' -o table > clusters.txt
+        clustername=$(tail -n 1 clusters.txt)
+        if [ "$clustername" == "" ]
+        then
+        echo AKS clusters unavailable
+        exit 1  
+        fi
+        echo $clustername
+        echo "##vso[task.setvariable variable=chosenclustername]$clustername"
+        echo 'az aks get-credentials -g $(AKS_CLUSTER_RG) -n $clustername'
```

Comment:
Where does this variable (AKS_CLUSTER_RG) come from? I think it's recommend to pass through variables using the "env" property so it is a little more explicit that it is coming from the pipeline. Can also help with debugging when you turn on "System Debugging" and re-run the pipeline.

Reply:
It does come from the pipeline variables. I documented these at the start of the pipeline